frontend/src/features/admin-form/create/builder-and-design/mutations/useCreateFormField.ts

```diff
@@ -54,13 +48,18 @@ export const useCreateFormField = () => {
         // Should not happen, should not be able to update field if there is no
         // existing data.
         if (!oldForm) throw new Error('Query should have been set')
-        oldForm.form_fields.splice(stateData.insertionIndex, 0, newField)
+        fieldBuilderStore.stateData.state === FieldBuilderState.CreatingField &&
```

Comment:
nit: prefer explicit conditionals, should we use an `if` statement here?

Reply:
updated